### PR TITLE
security fixes

### PR DESCRIPTION
**Pull request by mmaiers-nmdp**

### CODE REVIEW

**Review by jiyer-nmdp — APPROVED** [review]
These changes look good. Thanks Martin

**Comment by jiyer-nmdp** [comment]
Will wait for Pradeep to also review before merging

**Review by pbashyal-nmdp — APPROVED** [review]
👍

**Pull request merged**
